draft-ietf-quic-transport.md

```diff
@@ -1475,37 +1479,269 @@ failure.  If integrity protection is performed by QUIC, QUIC MUST abort the
+#### Responding to an Immediate Migration {#migration-response}
+
+Receiving a packet from a new client address indicates that the client may be
+immediately migrating if
+
+* the packet is a "non-probing" packet. That is, it contains a frame other than
+PATH_CHALLENGE, PATH_RESPONSE, and PADDING; and
+
+* the packet carries a packet number that is the largest seen thus far.
+
+Upon receiving such a packet, the server MUST abandon any path validation it is
+performing with other addresses on the expectation that those validations are
+likely to fail.  Abandoning path validation primarily means ceasing subsequent
+transmissions of the PATH_CHALLENGE frame.  An endpoint MAY ignore any
+subsequently received PATH_RESPONSE frames from abandoned addresses.
+
+In response to such a packet, the server MUST start sending subsequent packets
+to this client address and MUST initiate path validation to verify client
+ownership of the unvalidated address.
+
+A server MAY skip validation of a client address if it has been seen recently or
+if the server has reasonable certainty that the new address is the result of a
+NAT rebinding.
+
+Note that the server MAY send data to an unvalidated client address, but it MUST
+protect against potential attacks as described in {{address-spoofing}} and
+{{on-path-spoofing}}.
+
+If a non-probing packet is received from a new client address but with a packet
+number that is not the largest seen thus far, it is likely that it is a
+reordered packet sent from an older client address.  The server processes this
+packet as usual, but it MUST ignore this client address under the assumption
+that it is not the client's most recent address.
+
+After verifying a new client address, the server SHOULD send new address
+validation tokens ({{address-validation}}) to the client if the current ones
+will not be adequate to validate the new client address.
+
+
+### Handling Address Spoofing by a Client {#address-spoofing}
+
+It is possible that the client is spoofing its source address to cause the
+server to send excessive amounts of data to an unwilling host.  If the server
+sends significantly more data than the client, connection migration might be
+used to amplify the volume of data that an attacker can generate toward a
+victim.
+
+As described in {{migration-response}}, a server is required to validate the
+client's new address to confirm the client's possession of the new address.
+Until a client's address is deemed valid, the server MUST limit the rate at
+which it sends data to this address.  The server MUST NOT send more than a
+minimum congestion window's worth of data per estimated RTO period (as defined
+in {{QUIC-RECOVERY}}).  In the absence of this limit, the server risks being
+used for a denial of service attack against an unsuspecting victim.  Note that
+since the server will not have any round-trip time estimates to the new address,
+the RTO period is likely to be estimated based on the default initial RTT (see
```

Comment:
"SHOULD be" not "likely to be"

I think that this text belongs in the section that deals with handling of migrations by the non-migrating peer.

Reply:
done. I've simply moved this and the section below into the section above (Responding to Connection Migration).